### DISCUSSION TITLE
VSCode Extension GitHub Copilot could not connect to server

### DISCUSSION THREAD

**Discussion by gchli** [community/community#14305] [Copilot Conversations]
The error log is:
[ERROR] [default] [2022-04-05T03:23:30.634Z] GitHub Copilot could not connect to server. Extension activation failed: "Unexpected token N in JSON at position 0".

My vscode version is 1.66.0 and the copilot extension's version is 1.12.5517. I have tried a lot of methods such as reinstall this extension and reactivate this extension to solve this problem. But all of them can't work.
Could you tell me how does this happen and how can I solve this problem?
Thanks.

**Comment by davecheney** [comment]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Comment by small-handsome** [comment]


**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Comment by Wxh16144** [comment]
I'm having the same problem, which has occurred before, and inexplicably worked again in the middle. Now it appears once again. No idea how to troubleshoot the problem at all。

**Reply by davecheney** [reply]
Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Comment by zedzx1uv** [comment]
I am a student from SYSU in China and encounter the same issue. 
I modify the hosts as follows and it works:
140.82.112.5         api.github.com
140.82.112.5         github.com

**Reply by pizhn** [reply]
Works for me, thanks!

**Reply by VkingLangzi** [reply]
Works for me, thanks! +1

**Comment by krzychd** [comment]
Has same issue. It was related to my company proxy settings. I solved it using environment variable:
1st option (which worked for me):
Powershell:
`setx NO_PROXY "github.com,api.github.com"`

2nd option is to change settings proxy using cmd (run as admin):
`netsh winhttp show proxy`

you need to add "github.com;api.github.com;" at the end of your bypass list, example:
`netsh winhttp set proxy proxy-server="proxy_server_ip:port" bypass-list="*.microsoftonline.com;*.windows.net;…;*.g.nssvc.net;**github.com;api.github.com;**<local>"`
